File: src/my_robot_cpp/src/path_planner.cpp
#include "rclcpp/rclcpp.hpp"

class PathPlannerNode : public rclcpp::Node {
public:
    PathPlannerNode() : Node("path_planner_cpp") {
        RCLCPP_INFO(this->get_logger(), "C++ Path Planner started!");
        timer_ = this->create_wall_timer(
            std::chrono::seconds(1),
            std::bind(&PathPlannerNode::plan_path, this)
        );
    }

private:
    void plan_path() {
        RCLCPP_INFO(this->get_logger(), "Planning a path...");
    }
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<PathPlannerNode>());
    rclcpp::shutdown();
    return 0;
}

File: src/my_robot_py/my_robot_py/path_planner.py
import rclpy
from rclpy.node import Node

class PathPlannerNode(Node):
    def __init__(self):
        super().__init__('path_planner')
        self.get_logger().info('Path Planner node started!')
        
        # A timer that runs every 1 second
        self.timer = self.create_timer(1.0, self.plan_path)
    
    def plan_path(self):
        self.get_logger().info('Planning a path...')

def main(args=None):
    rclpy.init(args=args)
    node = PathPlannerNode()
    rclpy.spin(node)          # keeps the node alive
    rclpy.shutdown()

if __name__ == '__main__':
    main()